java/server/src/org/openqa/grid/common/RegistrationRequest.java

```diff
@@ -416,7 +416,13 @@ public static RegistrationRequest build(String... args) {
       try {
         URL ur = new URL(u);
         res.configuration.put(HUB_HOST, ur.getHost());
-        res.configuration.put(HUB_PORT, ur.getPort());
+        //If port was not defined after -hub default it to 4444
+        int port = ur.getPort();
+        if(port==-1){
+        	port=4444;
```

Comment:
forget it... keep it as is... :+1: i'll merge soon.  RemoteControlConfiguration is Selenium 1, and the less references we have to it, the better.